### PR TITLE
#2480 #2470 and unfiled crash in remote ?help

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#2480 Spaces placed around hyphen used for numeric negation			
#2470 Formatting: comments are not indented same as statement

Fix crash on remote ?help

### CODE REVIEW

File: src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs [hunk 1]
```diff
@@ -26,10 +26,9 @@ public class RemoteUriWebService : IRemoteUriWebService {
 
         public async Task GetResponseAsync(HttpListenerContext context, string localBaseUrl, string remoteBaseUrl, CancellationToken ct) {
             string postUri = null;
-            bool https = new Uri(remoteBaseUrl, UriKind.Absolute).IsHttps();
 
             if (context.Request.IsWebSocketRequest) {
-                UriBuilder ub = new UriBuilder(PostUri) { Scheme = https ? "wss" : "ws" };
```
Comment (int19h) [inline src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs]: Doesn't this regress local help? Or do we skip the URI service in that case?

**Comment by MikhailArkhipov** [comment]
It is not used in the local case

File: src/R/Core/Impl/Formatting/RFormatter.cs [hunk 2]
```diff
@@ -440,7 +440,10 @@ public class RFormatter {
         private void AppendOperator() {
             string text = _textProvider.GetText(_tokens.CurrentToken);
             if (_tokens.Position == 0 || TokenOperator.IsUnaryOperator(_tokens, _textProvider, TokenOperator.GetOperatorType(text), -1)) {
-                AppendToken(leadingSpace: true, trailingSpace: false);
+                // Determine if we need to insert leading space. We do in 'x + -1' but not in x + (-y + 1)
+                var precedingTokenType = _tokens.LookAhead(-1).TokenType;
+                bool leadingSpace = precedingTokenType == RTokenType.Operator;
```
Comment (int19h) [inline src/R/Core/Impl/Formatting/RFormatter.cs]: How does this handle cases where the preceding operator is also unary? E.g. `+-x` or `~-x`.

Also, for `x + -1`, wouldn't both spaces just come from the `+`?

**Comment by MikhailArkhipov** [comment]
We can technically walk back the chain until we find non-unary. Let me try that. As for spaces, in the text builder all operations are 'soft' so adding another space on top of existing one is no-op. Similarly, soft indent doesn't actually insert spaces/tabs until some non-ws sequence is inserted.

**Comment by int19h** [comment]
I see. But so long as all binary operators always add spaces around them, it sounds like we should just never add a space here at all, regardless of what precedes it? So far as I can see, it should always produce the right result without having to do any checks.

**Comment by MikhailArkhipov** [comment]
Good point

**Comment by int19h** [comment]
👍

**Pull request merged**